docs/llm_tools_invocation.md

```diff
@@ -0,0 +1,114 @@
+
+Errors must be raised as `HomeAssistantError` exceptions (or its subclasses). The response data should not contain error codes used for error handling.
+
+The `ToolInput` has following attributes:
+
+| Name              | Type    | Description                                                                                             |
+|-------------------|---------|---------------------------------------------------------------------------------------------------------|
+| `tool_name`       | string  | The name of the tool being called                                                                       |
+| `tool_args`       | dict    | The arguments provided by the LLM. The arguments are converted and validated using `parameters` schema. |
+| `platform`        | string  | The DOMAIN of the conversation agent using the tool                                                     |
+| `context`         | Context | The `homeassistant.core.Context` of the conversation                                                    |
+| `user_prompt`     | string  | The raw text input that initiated the tool call                                                         |
+| `language`        | string  | The language of the conversation agent, or "*" for any language                                         |
+| `assistant`       | string  | The assistant name used to control exposed entities. Currently only `conversation` is supported.        |
+
+## Getting available tools
+
+You can get the currently registered tools with `homeassistant.helpers.llm.async_get_tools(hass)`.
+
+If need the tool specification in OpenAPI-compatible format, you can use `voluptuous_openapi.convert` function (don't forget to mention the dependency in your `manifest.json` file).
+
+Example:
+```python
+from voluptuous_openapi import convert
+from homeassistant.helpers import llm
+...
+        tools = [
+            {
+                "name": tool.name,
+                "description": tool.description,
+                "parameters": convert(tool.parameters),
+            }
+            for tool in llm.async_get_tools(self.hass)
+        ]
+```
+
+Refer to the LLM API specification for the correct format.
+
+## Calling the tool
+
+When the LLM response indicates it wants to call a tool, you need to prepare the `llm.ToolInput` object, then execute the `llm.async_call_tool` function.
+
+Example:
+
+```python
+from homeassistant.helpers import llm
+
+class MyConversationEntity(
+    conversation.ConversationEntity, conversation.AbstractConversationAgent
+):
+    """Represent a conversation entity."""
+
+    async def async_process(
+        self, user_input: conversation.ConversationInput
+    ) -> conversation.ConversationResult:
+        """Process a sentence."""
+
+    while True:
+        response = ... # Get response from your LLM
```

Comment:
Perhaps say here in a comment:
- Add tools into the LLM call
- Or add previous tool result into the prompt